// @bootstrap-styled/css-utils/lib/float.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getFloatUtilities = getFloatUtilities;
exports.default = exports.defaultProps = void 0;

var _float = require("@bootstrap-styled/css-mixins/lib/float");

var _breakpoints = require("@bootstrap-styled/css-mixins/lib/breakpoints");

var defaultProps = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  }
};
exports.defaultProps = defaultProps;

function getFloatUtilities() {
  var gridBreakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$grid-breakpoints'];
  var floatUtilityList = [];
  Object.keys(gridBreakpoints).forEach(function (breakpoint) {
    var infix = (0, _breakpoints.breakpointInfix)(breakpoint, gridBreakpoints);
    var floatUtility = (0, _breakpoints.mediaBreakpointUp)(breakpoint, gridBreakpoints, "\n      .float".concat(infix, "-left {\n        ").concat((0, _float.floatLeft)(), "\n      }\n      .float").concat(infix, "-right {\n        ").concat((0, _float.floatRight)(), "\n      }\n      .float").concat(infix, "-none {\n        ").concat((0, _float.floatNone)(), "\n      }\n    "));
    floatUtilityList.push(floatUtility);
  });
  return floatUtilityList.join('\n');
}

var _default = {
  defaultProps: defaultProps,
  getFloatUtilities: getFloatUtilities
};
exports.default = _default;
