// @bootstrap-styled/css-utils/dist/@bootstrap-styled/css-utils.esm.js
function getAlignUtilities() {
  return "\n   ".concat(alignBaseline(), "\n   ").concat(alignTop(), "\n   ").concat(alignMiddle(), "\n   ").concat(alignBottom(), "\n   ").concat(alignTextBottom(), "\n   ").concat(alignTextTop(), "\n  ");
}
function alignBaseline() {
  return "\n    .align-baseline { vertical-align: baseline !important; } /* Browser default */\n  ";
}
function alignTop() {
  return "\n    .align-top { vertical-align: top !important; }\n  ";
}
function alignMiddle() {
  return "\n    .align-middle { vertical-align: middle !important; }\n  ";
}
function alignBottom() {
  return "\n    .align-bottom { vertical-align: bottom !important; }\n  ";
}
function alignTextBottom() {
  return "\n    .align-text-bottom { vertical-align: text-bottom !important; }\n  ";
}
function alignTextTop() {
  return "\n    .align-text-top { vertical-align: text-top !important; }\n  ";
}
var align = {
  getAlignUtilities: getAlignUtilities,
  alignBaseline: alignBaseline,
  alignTop: alignTop,
  alignMiddle: alignMiddle,
  alignBottom: alignBottom,
  alignTextBottom: alignTextBottom,
  alignTextTop: alignTextTop
};

function unwrapExports (x) {
	return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;
}

function createCommonjsModule(fn, module) {
	return module = { exports: {} }, fn(module, module.exports), module.exports;
}

function _classCallCheck(r,e){if(!(r instanceof e))throw new TypeError("Cannot call a class as a function")}var classCallCheck=_classCallCheck;function _defineProperty(r,e,n){return e in r?Object.defineProperty(r,e,{value:n,enumerable:!0,configurable:!0,writable:!0}):r[e]=n,r}var defineProperty=_defineProperty;function createCommonjsModule$1(r,e){return r(e={exports:{}},e.exports),e.exports}var colorName={aliceblue:[240,248,255],antiquewhite:[250,235,215],aqua:[0,255,255],aquamarine:[127,255,212],azure:[240,255,255],beige:[245,245,220],bisque:[255,228,196],black:[0,0,0],blanchedalmond:[255,235,205],blue:[0,0,255],blueviolet:[138,43,226],brown:[165,42,42],burlywood:[222,184,135],cadetblue:[95,158,160],chartreuse:[127,255,0],chocolate:[210,105,30],coral:[255,127,80],cornflowerblue:[100,149,237],cornsilk:[255,248,220],crimson:[220,20,60],cyan:[0,255,255],darkblue:[0,0,139],darkcyan:[0,139,139],darkgoldenrod:[184,134,11],darkgray:[169,169,169],darkgreen:[0,100,0],darkgrey:[169,169,169],darkkhaki:[189,183,107],darkmagenta:[139,0,139],darkolivegreen:[85,107,47],darkorange:[255,140,0],darkorchid:[153,50,204],darkred:[139,0,0],darksalmon:[233,150,122],darkseagreen:[143,188,143],darkslateblue:[72,61,139],darkslategray:[47,79,79],darkslategrey:[47,79,79],darkturquoise:[0,206,209],darkviolet:[148,0,211],deeppink:[255,20,147],deepskyblue:[0,191,255],dimgray:[105,105,105],dimgrey:[105,105,105],dodgerblue:[30,144,255],firebrick:[178,34,34],floralwhite:[255,250,240],forestgreen:[34,139,34],fuchsia:[255,0,255],gainsboro:[220,220,220],ghostwhite:[248,248,255],gold:[255,215,0],goldenrod:[218,165,32],gray:[128,128,128],green:[0,128,0],greenyellow:[173,255,47],grey:[128,128,128],honeydew:[240,255,240],hotpink:[255,105,180],indianred:[205,92,92],indigo:[75,0,130],ivory:[255,255,240],khaki:[240,230,140],lavender:[230,230,250],lavenderblush:[255,240,245],lawngreen:[124,252,0],lemonchiffon:[255,250,205],lightblue:[173,216,230],lightcoral:[240,128,128],lightcyan:[224,255,255],lightgoldenrodyellow:[250,250,210],lightgray:[211,211,211],lightgreen:[144,238,144],lightgrey:[211,211,211],lightpink:[255,182,193],lightsalmon:[255,160,122],lightseagreen:[32,178,170],lightskyblue:[135,206,250],lightslategray:[119,136,153],lightslategrey:[119,136,153],lightsteelblue:[176,196,222],lightyellow:[255,255,224],lime:[0,255,0],limegreen:[50,205,50],linen:[250,240,230],magenta:[255,0,255],maroon:[128,0,0],mediumaquamarine:[102,205,170],mediumblue:[0,0,205],mediumorchid:[186,85,211],mediumpurple:[147,112,219],mediumseagreen:[60,179,113],mediumslateblue:[123,104,238],mediumspringgreen:[0,250,154],mediumturquoise:[72,209,204],mediumvioletred:[199,21,133],midnightblue:[25,25,112],mintcream:[245,255,250],mistyrose:[255,228,225],moccasin:[255,228,181],navajowhite:[255,222,173],navy:[0,0,128],oldlace:[253,245,230],olive:[128,128,0],olivedrab:[107,142,35],orange:[255,165,0],orangered:[255,69,0],orchid:[218,112,214],palegoldenrod:[238,232,170],palegreen:[152,251,152],paleturquoise:[175,238,238],palevioletred:[219,112,147],papayawhip:[255,239,213],peachpuff:[255,218,185],peru:[205,133,63],pink:[255,192,203],plum:[221,160,221],powderblue:[176,224,230],purple:[128,0,128],rebeccapurple:[102,51,153],red:[255,0,0],rosybrown:[188,143,143],royalblue:[65,105,225],saddlebrown:[139,69,19],salmon:[250,128,114],sandybrown:[244,164,96],seagreen:[46,139,87],seashell:[255,245,238],sienna:[160,82,45],silver:[192,192,192],skyblue:[135,206,235],slateblue:[106,90,205],slategray:[112,128,144],slategrey:[112,128,144],snow:[255,250,250],springgreen:[0,255,127],steelblue:[70,130,180],tan:[210,180,140],teal:[0,128,128],thistle:[216,191,216],tomato:[255,99,71],turquoise:[64,224,208],violet:[238,130,238],wheat:[245,222,179],white:[255,255,255],whitesmoke:[245,245,245],yellow:[255,255,0],yellowgreen:[154,205,50]},isArrayish=function(r){return !(!r||"string"==typeof r)&&(r instanceof Array||Array.isArray(r)||r.length>=0&&(r.splice instanceof Function||Object.getOwnPropertyDescriptor(r,r.length-1)&&"String"!==r.constructor.name))},simpleSwizzle=createCommonjsModule$1(function(r){var e=Array.prototype.concat,n=Array.prototype.slice,t=r.exports=function(r){for(var t=[],o=0,a=r.length;o<a;o++){var i=r[o];isArrayish(i)?t=e.call(t,n.call(i)):t.push(i);}return t};t.wrap=function(r){return function(){return r(t(arguments))}};}),colorString=createCommonjsModule$1(function(r){var e={};for(var n in colorName)colorName.hasOwnProperty(n)&&(e[colorName[n]]=n);var t=r.exports={to:{},get:{}};function o(r,e,n){return Math.min(Math.max(e,r),n)}function a(r){var e=r.toString(16).toUpperCase();return e.length<2?"0"+e:e}t.get=function(r){var e,n;switch(r.substring(0,3).toLowerCase()){case"hsl":e=t.get.hsl(r),n="hsl";break;case"hwb":e=t.get.hwb(r),n="hwb";break;default:e=t.get.rgb(r),n="rgb";}return e?{model:n,value:e}:null},t.get.rgb=function(r){if(!r)return null;var e,n,t,a=[0,0,0,1];if(e=r.match(/^#([a-f0-9]{6})([a-f0-9]{2})?$/i)){for(t=e[2],e=e[1],n=0;n<3;n++){var i=2*n;a[n]=parseInt(e.slice(i,i+2),16);}t&&(a[3]=Math.round(parseInt(t,16)/255*100)/100);}else if(e=r.match(/^#([a-f0-9]{3,4})$/i)){for(t=(e=e[1])[3],n=0;n<3;n++)a[n]=parseInt(e[n]+e[n],16);t&&(a[3]=Math.round(parseInt(t+t,16)/255*100)/100);}else if(e=r.match(/^rgba?\(\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*,\s*([+-]?\d+)\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/)){for(n=0;n<3;n++)a[n]=parseInt(e[n+1],0);e[4]&&(a[3]=parseFloat(e[4]));}else {if(!(e=r.match(/^rgba?\(\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*,\s*([+-]?[\d\.]+)\%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/)))return (e=r.match(/(\D+)/))?"transparent"===e[1]?[0,0,0,0]:(a=colorName[e[1]])?(a[3]=1,a):null:null;for(n=0;n<3;n++)a[n]=Math.round(2.55*parseFloat(e[n+1]));e[4]&&(a[3]=parseFloat(e[4]));}for(n=0;n<3;n++)a[n]=o(a[n],0,255);return a[3]=o(a[3],0,1),a},t.get.hsl=function(r){if(!r)return null;var e=r.match(/^hsla?\(\s*([+-]?(?:\d*\.)?\d+)(?:deg)?\s*,\s*([+-]?[\d\.]+)%\s*,\s*([+-]?[\d\.]+)%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/);if(e){var n=parseFloat(e[4]);return [(parseFloat(e[1])+360)%360,o(parseFloat(e[2]),0,100),o(parseFloat(e[3]),0,100),o(isNaN(n)?1:n,0,1)]}return null},t.get.hwb=function(r){if(!r)return null;var e=r.match(/^hwb\(\s*([+-]?\d*[\.]?\d+)(?:deg)?\s*,\s*([+-]?[\d\.]+)%\s*,\s*([+-]?[\d\.]+)%\s*(?:,\s*([+-]?[\d\.]+)\s*)?\)$/);if(e){var n=parseFloat(e[4]);return [(parseFloat(e[1])%360+360)%360,o(parseFloat(e[2]),0,100),o(parseFloat(e[3]),0,100),o(isNaN(n)?1:n,0,1)]}return null},t.to.hex=function(){var r=simpleSwizzle(arguments);return "#"+a(r[0])+a(r[1])+a(r[2])+(r[3]<1?a(Math.round(255*r[3])):"")},t.to.rgb=function(){var r=simpleSwizzle(arguments);return r.length<4||1===r[3]?"rgb("+Math.round(r[0])+", "+Math.round(r[1])+", "+Math.round(r[2])+")":"rgba("+Math.round(r[0])+", "+Math.round(r[1])+", "+Math.round(r[2])+", "+r[3]+")"},t.to.rgb.percent=function(){var r=simpleSwizzle(arguments),e=Math.round(r[0]/255*100),n=Math.round(r[1]/255*100),t=Math.round(r[2]/255*100);return r.length<4||1===r[3]?"rgb("+e+"%, "+n+"%, "+t+"%)":"rgba("+e+"%, "+n+"%, "+t+"%, "+r[3]+")"},t.to.hsl=function(){var r=simpleSwizzle(arguments);return r.length<4||1===r[3]?"hsl("+r[0]+", "+r[1]+"%, "+r[2]+"%)":"hsla("+r[0]+", "+r[1]+"%, "+r[2]+"%, "+r[3]+")"},t.to.hwb=function(){var r=simpleSwizzle(arguments),e="";return r.length>=4&&1!==r[3]&&(e=", "+r[3]),"hwb("+r[0]+", "+r[1]+"%, "+r[2]+"%"+e+")"},t.to.keyword=function(r){return e[r.slice(0,3)]};}),colorString_1=colorString.to,colorString_2=colorString.get,conversions=createCommonjsModule$1(function(r){var e={};for(var n in colorName)colorName.hasOwnProperty(n)&&(e[colorName[n]]=n);var t=r.exports={rgb:{channels:3,labels:"rgb"},hsl:{channels:3,labels:"hsl"},hsv:{channels:3,labels:"hsv"},hwb:{channels:3,labels:"hwb"},cmyk:{channels:4,labels:"cmyk"},xyz:{channels:3,labels:"xyz"},lab:{channels:3,labels:"lab"},lch:{channels:3,labels:"lch"},hex:{channels:1,labels:["hex"]},keyword:{channels:1,labels:["keyword"]},ansi16:{channels:1,labels:["ansi16"]},ansi256:{channels:1,labels:["ansi256"]},hcg:{channels:3,labels:["h","c","g"]},apple:{channels:3,labels:["r16","g16","b16"]},gray:{channels:1,labels:["gray"]}};for(var o in t)if(t.hasOwnProperty(o)){if(!("channels"in t[o]))throw new Error("missing channels property: "+o);if(!("labels"in t[o]))throw new Error("missing channel labels property: "+o);if(t[o].labels.length!==t[o].channels)throw new Error("channel and label counts mismatch: "+o);var a=t[o].channels,i=t[o].labels;delete t[o].channels,delete t[o].labels,Object.defineProperty(t[o],"channels",{value:a}),Object.defineProperty(t[o],"labels",{value:i});}t.rgb.hsl=function(r){var e,n,t=r[0]/255,o=r[1]/255,a=r[2]/255,i=Math.min(t,o,a),s=Math.max(t,o,a),l=s-i;return s===i?e=0:t===s?e=(o-a)/l:o===s?e=2+(a-t)/l:a===s&&(e=4+(t-o)/l),(e=Math.min(60*e,360))<0&&(e+=360),n=(i+s)/2,[e,100*(s===i?0:n<=.5?l/(s+i):l/(2-s-i)),100*n]},t.rgb.hsv=function(r){var e,n,t,o,a,i=r[0]/255,s=r[1]/255,l=r[2]/255,c=Math.max(i,s,l),h=c-Math.min(i,s,l),u=function(r){return (c-r)/6/h+.5};return 0===h?o=a=0:(a=h/c,e=u(i),n=u(s),t=u(l),i===c?o=t-n:s===c?o=1/3+e-t:l===c&&(o=2/3+n-e),o<0?o+=1:o>1&&(o-=1)),[360*o,100*a,100*c]},t.rgb.hwb=function(r){var e=r[0],n=r[1],o=r[2];return [t.rgb.hsl(r)[0],100*(1/255*Math.min(e,Math.min(n,o))),100*(o=1-1/255*Math.max(e,Math.max(n,o)))]},t.rgb.cmyk=function(r){var e,n=r[0]/255,t=r[1]/255,o=r[2]/255;return [100*((1-n-(e=Math.min(1-n,1-t,1-o)))/(1-e)||0),100*((1-t-e)/(1-e)||0),100*((1-o-e)/(1-e)||0),100*e]},t.rgb.keyword=function(r){var n=e[r];if(n)return n;var t,o,a,i=1/0;for(var s in colorName)if(colorName.hasOwnProperty(s)){var l=colorName[s],c=(o=r,a=l,Math.pow(o[0]-a[0],2)+Math.pow(o[1]-a[1],2)+Math.pow(o[2]-a[2],2));c<i&&(i=c,t=s);}return t},t.keyword.rgb=function(r){return colorName[r]},t.rgb.xyz=function(r){var e=r[0]/255,n=r[1]/255,t=r[2]/255;return [100*(.4124*(e=e>.04045?Math.pow((e+.055)/1.055,2.4):e/12.92)+.3576*(n=n>.04045?Math.pow((n+.055)/1.055,2.4):n/12.92)+.1805*(t=t>.04045?Math.pow((t+.055)/1.055,2.4):t/12.92)),100*(.2126*e+.7152*n+.0722*t),100*(.0193*e+.1192*n+.9505*t)]},t.rgb.lab=function(r){var e=t.rgb.xyz(r),n=e[0],o=e[1],a=e[2];return o/=100,a/=108.883,n=(n/=95.047)>.008856?Math.pow(n,1/3):7.787*n+16/116,[116*(o=o>.008856?Math.pow(o,1/3):7.787*o+16/116)-16,500*(n-o),200*(o-(a=a>.008856?Math.pow(a,1/3):7.787*a+16/116))]},t.hsl.rgb=function(r){var e,n,t,o,a,i=r[0]/360,s=r[1]/100,l=r[2]/100;if(0===s)return [a=255*l,a,a];e=2*l-(n=l<.5?l*(1+s):l+s-l*s),o=[0,0,0];for(var c=0;c<3;c++)(t=i+1/3*-(c-1))<0&&t++,t>1&&t--,a=6*t<1?e+6*(n-e)*t:2*t<1?n:3*t<2?e+(n-e)*(2/3-t)*6:e,o[c]=255*a;return o},t.hsl.hsv=function(r){var e=r[0],n=r[1]/100,t=r[2]/100,o=n,a=Math.max(t,.01);return n*=(t*=2)<=1?t:2-t,o*=a<=1?a:2-a,[e,100*(0===t?2*o/(a+o):2*n/(t+n)),100*((t+n)/2)]},t.hsv.rgb=function(r){var e=r[0]/60,n=r[1]/100,t=r[2]/100,o=Math.floor(e)%6,a=e-Math.floor(e),i=255*t*(1-n),s=255*t*(1-n*a),l=255*t*(1-n*(1-a));switch(t*=255,o){case 0:return [t,l,i];case 1:return [s,t,i];case 2:return [i,t,l];case 3:return [i,s,t];case 4:return [l,i,t];case 5:return [t,i,s]}},t.hsv.hsl=function(r){var e,n,t,o=r[0],a=r[1]/100,i=r[2]/100,s=Math.max(i,.01);return t=(2-a)*i,n=a*s,[o,100*(n=(n/=(e=(2-a)*s)<=1?e:2-e)||0),100*(t/=2)]},t.hwb.rgb=function(r){var e,n,t,o,a,i,s,l=r[0]/360,c=r[1]/100,h=r[2]/100,u=c+h;switch(u>1&&(c/=u,h/=u),t=6*l-(e=Math.floor(6*l)),0!=(1&e)&&(t=1-t),o=c+t*((n=1-h)-c),e){default:case 6:case 0:a=n,i=o,s=c;break;case 1:a=o,i=n,s=c;break;case 2:a=c,i=n,s=o;break;case 3:a=c,i=o,s=n;break;case 4:a=o,i=c,s=n;break;case 5:a=n,i=c,s=o;}return [255*a,255*i,255*s]},t.cmyk.rgb=function(r){var e=r[0]/100,n=r[1]/100,t=r[2]/100,o=r[3]/100;return [255*(1-Math.min(1,e*(1-o)+o)),255*(1-Math.min(1,n*(1-o)+o)),255*(1-Math.min(1,t*(1-o)+o))]},t.xyz.rgb=function(r){var e,n,t,o=r[0]/100,a=r[1]/100,i=r[2]/100;return n=-.9689*o+1.8758*a+.0415*i,t=.0557*o+-.204*a+1.057*i,e=(e=3.2406*o+-1.5372*a+-.4986*i)>.0031308?1.055*Math.pow(e,1/2.4)-.055:12.92*e,n=n>.0031308?1.055*Math.pow(n,1/2.4)-.055:12.92*n,t=t>.0031308?1.055*Math.pow(t,1/2.4)-.055:12.92*t,[255*(e=Math.min(Math.max(0,e),1)),255*(n=Math.min(Math.max(0,n),1)),255*(t=Math.min(Math.max(0,t),1))]},t.xyz.lab=function(r){var e=r[0],n=r[1],t=r[2];return n/=100,t/=108.883,e=(e/=95.047)>.008856?Math.pow(e,1/3):7.787*e+16/116,[116*(n=n>.008856?Math.pow(n,1/3):7.787*n+16/116)-16,500*(e-n),200*(n-(t=t>.008856?Math.pow(t,1/3):7.787*t+16/116))]},t.lab.xyz=function(r){var e,n,t,o=r[0];e=r[1]/500+(n=(o+16)/116),t=n-r[2]/200;var a=Math.pow(n,3),i=Math.pow(e,3),s=Math.pow(t,3);return n=a>.008856?a:(n-16/116)/7.787,e=i>.008856?i:(e-16/116)/7.787,t=s>.008856?s:(t-16/116)/7.787,[e*=95.047,n*=100,t*=108.883]},t.lab.lch=function(r){var e,n=r[0],t=r[1],o=r[2];return (e=360*Math.atan2(o,t)/2/Math.PI)<0&&(e+=360),[n,Math.sqrt(t*t+o*o),e]},t.lch.lab=function(r){var e,n=r[0],t=r[1];return e=r[2]/360*2*Math.PI,[n,t*Math.cos(e),t*Math.sin(e)]},t.rgb.ansi16=function(r){var e=r[0],n=r[1],o=r[2],a=1 in arguments?arguments[1]:t.rgb.hsv(r)[2];if(0===(a=Math.round(a/50)))return 30;var i=30+(Math.round(o/255)<<2|Math.round(n/255)<<1|Math.round(e/255));return 2===a&&(i+=60),i},t.hsv.ansi16=function(r){return t.rgb.ansi16(t.hsv.rgb(r),r[2])},t.rgb.ansi256=function(r){var e=r[0],n=r[1],t=r[2];return e===n&&n===t?e<8?16:e>248?231:Math.round((e-8)/247*24)+232:16+36*Math.round(e/255*5)+6*Math.round(n/255*5)+Math.round(t/255*5)},t.ansi16.rgb=function(r){var e=r%10;if(0===e||7===e)return r>50&&(e+=3.5),[e=e/10.5*255,e,e];var n=.5*(1+~~(r>50));return [(1&e)*n*255,(e>>1&1)*n*255,(e>>2&1)*n*255]},t.ansi256.rgb=function(r){if(r>=232){var e=10*(r-232)+8;return [e,e,e]}var n;return r-=16,[Math.floor(r/36)/5*255,Math.floor((n=r%36)/6)/5*255,n%6/5*255]},t.rgb.hex=function(r){var e=(((255&Math.round(r[0]))<<16)+((255&Math.round(r[1]))<<8)+(255&Math.round(r[2]))).toString(16).toUpperCase();return "000000".substring(e.length)+e},t.hex.rgb=function(r){var e=r.toString(16).match(/[a-f0-9]{6}|[a-f0-9]{3}/i);if(!e)return [0,0,0];var n=e[0];3===e[0].length&&(n=n.split("").map(function(r){return r+r}).join(""));var t=parseInt(n,16);return [t>>16&255,t>>8&255,255&t]},t.rgb.hcg=function(r){var e,n=r[0]/255,t=r[1]/255,o=r[2]/255,a=Math.max(Math.max(n,t),o),i=Math.min(Math.min(n,t),o),s=a-i;return e=s<=0?0:a===n?(t-o)/s%6:a===t?2+(o-n)/s:4+(n-t)/s+4,e/=6,[360*(e%=1),100*s,100*(s<1?i/(1-s):0)]},t.hsl.hcg=function(r){var e=r[1]/100,n=r[2]/100,t=1,o=0;return (t=n<.5?2*e*n:2*e*(1-n))<1&&(o=(n-.5*t)/(1-t)),[r[0],100*t,100*o]},t.hsv.hcg=function(r){var e=r[1]/100,n=r[2]/100,t=e*n,o=0;return t<1&&(o=(n-t)/(1-t)),[r[0],100*t,100*o]},t.hcg.rgb=function(r){var e=r[0]/360,n=r[1]/100,t=r[2]/100;if(0===n)return [255*t,255*t,255*t];var o,a=[0,0,0],i=e%1*6,s=i%1,l=1-s;switch(Math.floor(i)){case 0:a[0]=1,a[1]=s,a[2]=0;break;case 1:a[0]=l,a[1]=1,a[2]=0;break;case 2:a[0]=0,a[1]=1,a[2]=s;break;case 3:a[0]=0,a[1]=l,a[2]=1;break;case 4:a[0]=s,a[1]=0,a[2]=1;break;default:a[0]=1,a[1]=0,a[2]=l;}return o=(1-n)*t,[255*(n*a[0]+o),255*(n*a[1]+o),255*(n*a[2]+o)]},t.hcg.hsv=function(r){var e=r[1]/100,n=e+r[2]/100*(1-e),t=0;return n>0&&(t=e/n),[r[0],100*t,100*n]},t.hcg.hsl=function(r){var e=r[1]/100,n=r[2]/100*(1-e)+.5*e,t=0;return n>0&&n<.5?t=e/(2*n):n>=.5&&n<1&&(t=e/(2*(1-n))),[r[0],100*t,100*n]},t.hcg.hwb=function(r){var e=r[1]/100,n=e+r[2]/100*(1-e);return [r[0],100*(n-e),100*(1-n)]},t.hwb.hcg=function(r){var e=r[1]/100,n=1-r[2]/100,t=n-e,o=0;return t<1&&(o=(n-t)/(1-t)),[r[0],100*t,100*o]},t.apple.rgb=function(r){return [r[0]/65535*255,r[1]/65535*255,r[2]/65535*255]},t.rgb.apple=function(r){return [r[0]/255*65535,r[1]/255*65535,r[2]/255*65535]},t.gray.rgb=function(r){return [r[0]/100*255,r[0]/100*255,r[0]/100*255]},t.gray.hsl=t.gray.hsv=function(r){return [0,0,r[0]]},t.gray.hwb=function(r){return [0,100,r[0]]},t.gray.cmyk=function(r){return [0,0,0,r[0]]},t.gray.lab=function(r){return [r[0],0,0]},t.gray.hex=function(r){var e=255&Math.round(r[0]/100*255),n=((e<<16)+(e<<8)+e).toString(16).toUpperCase();return "000000".substring(n.length)+n},t.rgb.gray=function(r){return [(r[0]+r[1]+r[2])/3/255*100]};}),conversions_1=conversions.rgb,conversions_2=conversions.hsl,conversions_3=conversions.hsv,conversions_4=conversions.hwb,conversions_5=conversions.cmyk,conversions_6=conversions.xyz,conversions_7=conversions.lab,conversions_8=conversions.lch,conversions_9=conversions.hex,conversions_10=conversions.keyword,conversions_11=conversions.ansi16,conversions_12=conversions.ansi256,conversions_13=conversions.hcg,conversions_14=conversions.apple,conversions_15=conversions.gray;function buildGraph(){for(var r={},e=Object.keys(conversions),n=e.length,t=0;t<n;t++)r[e[t]]={distance:-1,parent:null};return r}function deriveBFS(r){var e=buildGraph(),n=[r];for(e[r].distance=0;n.length;)for(var t=n.pop(),o=Object.keys(conversions[t]),a=o.length,i=0;i<a;i++){var s=o[i],l=e[s];-1===l.distance&&(l.distance=e[t].distance+1,l.parent=t,n.unshift(s));}return e}function link(r,e){return function(n){return e(r(n))}}function wrapConversion(r,e){for(var n=[e[r].parent,r],t=conversions[e[r].parent][r],o=e[r].parent;e[o].parent;)n.unshift(e[o].parent),t=link(conversions[e[o].parent][o],t),o=e[o].parent;return t.conversion=n,t}var route=function(r){for(var e=deriveBFS(r),n={},t=Object.keys(e),o=t.length,a=0;a<o;a++){var i=t[a];null!==e[i].parent&&(n[i]=wrapConversion(i,e));}return n},convert={},models=Object.keys(conversions);function wrapRaw(r){var e=function(e){return null==e?e:(arguments.length>1&&(e=Array.prototype.slice.call(arguments)),r(e))};return "conversion"in r&&(e.conversion=r.conversion),e}function wrapRounded(r){var e=function(e){if(null==e)return e;arguments.length>1&&(e=Array.prototype.slice.call(arguments));var n=r(e);if("object"==typeof n)for(var t=n.length,o=0;o<t;o++)n[o]=Math.round(n[o]);return n};return "conversion"in r&&(e.conversion=r.conversion),e}models.forEach(function(r){convert[r]={},Object.defineProperty(convert[r],"channels",{value:conversions[r].channels}),Object.defineProperty(convert[r],"labels",{value:conversions[r].labels});var e=route(r);Object.keys(e).forEach(function(n){var t=e[n];convert[r][n]=wrapRounded(t),convert[r][n].raw=wrapRaw(t);});});var colorConvert=convert,_slice=[].slice,skippedModels=["keyword","gray","hex"],hashedModelKeys={};Object.keys(colorConvert).forEach(function(r){hashedModelKeys[_slice.call(colorConvert[r].labels).sort().join("")]=r;});var limiters={};function Color(r,e){if(!(this instanceof Color))return new Color(r,e);if(e&&e in skippedModels&&(e=null),e&&!(e in colorConvert))throw new Error("Unknown model: "+e);var n,t;if(void 0===r)this.model="rgb",this.color=[0,0,0],this.valpha=1;else if(r instanceof Color)this.model=r.model,this.color=r.color.slice(),this.valpha=r.valpha;else if("string"==typeof r){var o=colorString.get(r);if(null===o)throw new Error("Unable to parse color from string: "+r);this.model=o.model,t=colorConvert[this.model].channels,this.color=o.value.slice(0,t),this.valpha="number"==typeof o.value[t]?o.value[t]:1;}else if(r.length){this.model=e||"rgb",t=colorConvert[this.model].channels;var a=_slice.call(r,0,t);this.color=zeroArray(a,t),this.valpha="number"==typeof r[t]?r[t]:1;}else if("number"==typeof r)r&=16777215,this.model="rgb",this.color=[r>>16&255,r>>8&255,255&r],this.valpha=1;else {this.valpha=1;var i=Object.keys(r);"alpha"in r&&(i.splice(i.indexOf("alpha"),1),this.valpha="number"==typeof r.alpha?r.alpha:0);var s=i.sort().join("");if(!(s in hashedModelKeys))throw new Error("Unable to parse color from object: "+JSON.stringify(r));this.model=hashedModelKeys[s];var l=colorConvert[this.model].labels,c=[];for(n=0;n<l.length;n++)c.push(r[l[n]]);this.color=zeroArray(c);}if(limiters[this.model])for(t=colorConvert[this.model].channels,n=0;n<t;n++){var h=limiters[this.model][n];h&&(this.color[n]=h(this.color[n]));}this.valpha=Math.max(0,Math.min(1,this.valpha)),Object.freeze&&Object.freeze(this);}function roundTo(r,e){return Number(r.toFixed(e))}function roundToPlace(r){return function(e){return roundTo(e,r)}}function getset(r,e,n){return (r=Array.isArray(r)?r:[r]).forEach(function(r){(limiters[r]||(limiters[r]=[]))[e]=n;}),r=r[0],function(t){var o;return arguments.length?(n&&(t=n(t)),(o=this[r]()).color[e]=t,o):(o=this[r]().color[e],n&&(o=n(o)),o)}}function maxfn(r){return function(e){return Math.max(0,Math.min(r,e))}}function assertArray(r){return Array.isArray(r)?r:[r]}function zeroArray(r,e){for(var n=0;n<e;n++)"number"!=typeof r[n]&&(r[n]=0);return r}Color.prototype={toString:function(){return this.string()},toJSON:function(){return this[this.model]()},string:function(r){var e=this.model in colorString.to?this:this.rgb(),n=1===(e=e.round("number"==typeof r?r:1)).valpha?e.color:e.color.concat(this.valpha);return colorString.to[e.model](n)},percentString:function(r){var e=this.rgb().round("number"==typeof r?r:1),n=1===e.valpha?e.color:e.color.concat(this.valpha);return colorString.to.rgb.percent(n)},array:function(){return 1===this.valpha?this.color.slice():this.color.concat(this.valpha)},object:function(){for(var r={},e=colorConvert[this.model].channels,n=colorConvert[this.model].labels,t=0;t<e;t++)r[n[t]]=this.color[t];return 1!==this.valpha&&(r.alpha=this.valpha),r},unitArray:function(){var r=this.rgb().color;return r[0]/=255,r[1]/=255,r[2]/=255,1!==this.valpha&&r.push(this.valpha),r},unitObject:function(){var r=this.rgb().object();return r.r/=255,r.g/=255,r.b/=255,1!==this.valpha&&(r.alpha=this.valpha),r},round:function(r){return r=Math.max(r||0,0),new Color(this.color.map(roundToPlace(r)).concat(this.valpha),this.model)},alpha:function(r){return arguments.length?new Color(this.color.concat(Math.max(0,Math.min(1,r))),this.model):this.valpha},red:getset("rgb",0,maxfn(255)),green:getset("rgb",1,maxfn(255)),blue:getset("rgb",2,maxfn(255)),hue:getset(["hsl","hsv","hsl","hwb","hcg"],0,function(r){return (r%360+360)%360}),saturationl:getset("hsl",1,maxfn(100)),lightness:getset("hsl",2,maxfn(100)),saturationv:getset("hsv",1,maxfn(100)),value:getset("hsv",2,maxfn(100)),chroma:getset("hcg",1,maxfn(100)),gray:getset("hcg",2,maxfn(100)),white:getset("hwb",1,maxfn(100)),wblack:getset("hwb",2,maxfn(100)),cyan:getset("cmyk",0,maxfn(100)),magenta:getset("cmyk",1,maxfn(100)),yellow:getset("cmyk",2,maxfn(100)),black:getset("cmyk",3,maxfn(100)),x:getset("xyz",0,maxfn(100)),y:getset("xyz",1,maxfn(100)),z:getset("xyz",2,maxfn(100)),l:getset("lab",0,maxfn(100)),a:getset("lab",1),b:getset("lab",2),keyword:function(r){return arguments.length?new Color(r):colorConvert[this.model].keyword(this.color)},hex:function(r){return arguments.length?new Color(r):colorString.to.hex(this.rgb().round().color)},rgbNumber:function(){var r=this.rgb().color;return (255&r[0])<<16|(255&r[1])<<8|255&r[2]},luminosity:function(){for(var r=this.rgb().color,e=[],n=0;n<r.length;n++){var t=r[n]/255;e[n]=t<=.03928?t/12.92:Math.pow((t+.055)/1.055,2.4);}return .2126*e[0]+.7152*e[1]+.0722*e[2]},contrast:function(r){var e=this.luminosity(),n=r.luminosity();return e>n?(e+.05)/(n+.05):(n+.05)/(e+.05)},level:function(r){var e=this.contrast(r);return e>=7.1?"AAA":e>=4.5?"AA":""},isDark:function(){var r=this.rgb().color;return (299*r[0]+587*r[1]+114*r[2])/1e3<128},isLight:function(){return !this.isDark()},negate:function(){for(var r=this.rgb(),e=0;e<3;e++)r.color[e]=255-r.color[e];return r},lighten:function(r){var e=this.hsl();return e.color[2]+=e.color[2]*r,e},darken:function(r){var e=this.hsl();return e.color[2]-=e.color[2]*r,e},saturate:function(r){var e=this.hsl();return e.color[1]+=e.color[1]*r,e},desaturate:function(r){var e=this.hsl();return e.color[1]-=e.color[1]*r,e},whiten:function(r){var e=this.hwb();return e.color[1]+=e.color[1]*r,e},blacken:function(r){var e=this.hwb();return e.color[2]+=e.color[2]*r,e},grayscale:function(){var r=this.rgb().color,e=.3*r[0]+.59*r[1]+.11*r[2];return Color.rgb(e,e,e)},fade:function(r){return this.alpha(this.valpha-this.valpha*r)},opaquer:function(r){return this.alpha(this.valpha+this.valpha*r)},rotate:function(r){var e=this.hsl(),n=e.color[0];return n=(n=(n+r)%360)<0?360+n:n,e.color[0]=n,e},mix:function(r,e){var n=r.rgb(),t=this.rgb(),o=void 0===e?.5:e,a=2*o-1,i=n.alpha()-t.alpha(),s=((a*i==-1?a:(a+i)/(1+a*i))+1)/2,l=1-s;return Color.rgb(s*n.red()+l*t.red(),s*n.green()+l*t.green(),s*n.blue()+l*t.blue(),n.alpha()*o+t.alpha()*(1-o))}},Object.keys(colorConvert).forEach(function(r){if(-1===skippedModels.indexOf(r)){var e=colorConvert[r].channels;Color.prototype[r]=function(){if(this.model===r)return new Color(this);if(arguments.length)return new Color(arguments,r);var n="number"==typeof arguments[e]?e:this.valpha;return new Color(assertArray(colorConvert[this.model][r].raw(this.color)).concat(n),r)},Color[r]=function(n){return "number"==typeof n&&(n=zeroArray(_slice.call(arguments),e)),new Color(n,r)};}});var color=Color,ColorMock=function r(e){var n=this;classCallCheck(this,r),defineProperty(this,"hsl",function(){return n}),defineProperty(this,"isLight",function(){return !1}),defineProperty(this,"isDark",function(){return !1}),defineProperty(this,"negate",function(){return n}),defineProperty(this,"lighten",function(){return n}),defineProperty(this,"darken",function(){return n}),defineProperty(this,"alpha",function(){return n}),defineProperty(this,"saturate",function(){return n}),defineProperty(this,"desaturate",function(){return n}),defineProperty(this,"grayscale",function(){return n}),defineProperty(this,"whiten",function(){return n}),defineProperty(this,"blacken",function(){return n}),defineProperty(this,"fade",function(){return n}),defineProperty(this,"opaquer",function(){return n}),defineProperty(this,"rotate",function(){return n}),defineProperty(this,"contrast",function(){return 10}),defineProperty(this,"luminosity",function(){return .5}),defineProperty(this,"mix",function(){return n}),defineProperty(this,"blue",function(){return n.toString()}),defineProperty(this,"green",function(){return n.toString()}),defineProperty(this,"red",function(){return n.toString()}),defineProperty(this,"hex",function(){return n.toString()}),defineProperty(this,"rgbNumber",function(){return n.toString()}),defineProperty(this,"rgb",function(){return n.toString()}),defineProperty(this,"toString",function(){return n.color}),this.color=e;};function ColorWrapper(r){return "string"==typeof r&&-1!==r.indexOf("linear-gradient")?new ColorMock(r):color(r)}

var MixinError_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }
function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }
function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }
function _wrapNativeSuper(Class) { var _cache = typeof Map === "function" ? new Map() : undefined; _wrapNativeSuper = function _wrapNativeSuper(Class) { if (Class === null || !_isNativeFunction(Class)) return Class; if (typeof Class !== "function") { throw new TypeError("Super expression must either be null or a function"); } if (typeof _cache !== "undefined") { if (_cache.has(Class)) return _cache.get(Class); _cache.set(Class, Wrapper); } function Wrapper() { return _construct(Class, arguments, _getPrototypeOf(this).constructor); } Wrapper.prototype = Object.create(Class.prototype, { constructor: { value: Wrapper, enumerable: false, writable: true, configurable: true } }); return _setPrototypeOf(Wrapper, Class); }; return _wrapNativeSuper(Class); }
function isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }
function _construct(Parent, args, Class) { if (isNativeReflectConstruct()) { _construct = Reflect.construct; } else { _construct = function _construct(Parent, args, Class) { var a = [null]; a.push.apply(a, args); var Constructor = Function.bind.apply(Parent, a); var instance = new Constructor(); if (Class) _setPrototypeOf(instance, Class.prototype); return instance; }; } return _construct.apply(null, arguments); }
function _isNativeFunction(fn) { return Function.toString.call(fn).indexOf("[native code]") !== -1; }
function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }
function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }
var MixinError =
function (_Error) {
  _inherits(MixinError, _Error);
  function MixinError(args) {
    var _this;
    _classCallCheck(this, MixinError);
    _this = _possibleConstructorReturn(this, _getPrototypeOf(MixinError).call(this, args));
    Error.captureStackTrace(_assertThisInitialized(_this), MixinError);
    return _this;
  }
  return MixinError;
}(_wrapNativeSuper(Error));
exports.default = MixinError;
module.exports = exports.default;
});
unwrapExports(MixinError_1);

var hover_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.hover = hover;
exports.hoverFocus = hoverFocus;
exports.plainHoverFocus = plainHoverFocus;
exports.hoverFocusActive = hoverFocusActive;
exports.default = exports.defaultProps = void 0;
var _MixinError = _interopRequireDefault(MixinError_1);
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var defaultProps = {
  '$enable-hover-media-query': false
};
exports.defaultProps = defaultProps;
function hover(content) {
  if (!content) throw new _MixinError.default('content is required');
  return "&:hover, &.hover { ".concat(content, " }");
}
function hoverFocus() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var content = arguments.length > 1 ? arguments[1] : undefined;
  if (enableHoverMediaQuery) {
    return " \n      &:focus, &.focus { ".concat(content, " }\n      ").concat(hover(content), "\n    ");
  }
  return "\n    &:focus,\n    &.focus,\n    &:hover,\n    &.hover {\n      ".concat(content, "\n    }\n  ");
}
function plainHoverFocus() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var content = arguments.length > 1 ? arguments[1] : undefined;
  if (enableHoverMediaQuery) {
    return "\n      &, \n      &:focus, \n      &.focus {\n        ".concat(content, "\n      }\n      ").concat(hover(content), "\n    ");
  }
  return " \n    &, \n    &:focus, \n    &.focus, \n    &:hover, \n    &.hover {\n      ".concat(content, "\n    }\n  ");
}
function hoverFocusActive() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var content = arguments.length > 1 ? arguments[1] : undefined;
  if (enableHoverMediaQuery) {
    return "\n      &:focus,\n      &.focus,\n      &:active,\n      &.active {\n        ".concat(content, "\n      }\n      ").concat(hover(content), "\n    ");
  }
  return "\n    &:focus, \n    &.focus, \n    &:active, \n    &.active,\n    &:hover,\n    &.hover {\n     ".concat(content, "\n    }\n  ");
}
hover.focus = hoverFocus;
hover.plainFocus = plainHoverFocus;
hover.activeFocus = hoverFocusActive;
var _default = {
  defaultProps: defaultProps,
  hover: hover,
  hoverFocus: hoverFocus,
  plainHoverFocus: plainHoverFocus,
  hoverFocusActive: hoverFocusActive
};
exports.default = _default;
});
unwrapExports(hover_1);
var hover_2 = hover_1.hover;
var hover_3 = hover_1.hoverFocus;
var hover_4 = hover_1.plainHoverFocus;
var hover_5 = hover_1.hoverFocusActive;
var hover_6 = hover_1.defaultProps;

var backgroundVariant = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.bgVariant = bgVariant;
exports.default = exports.defaultProps = void 0;
var _color = _interopRequireDefault(ColorWrapper);
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var defaultProps = {
  '$enable-hover-media-query': false
};
exports.defaultProps = defaultProps;
function bgVariant() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var selector = arguments.length > 1 ? arguments[1] : undefined;
  var bgColor = arguments.length > 2 ? arguments[2] : undefined;
  return "\n    ".concat(selector, " {\n      background-color: ").concat(bgColor, " !important;\n    }\n    a").concat(selector, " {\n      ").concat((0, hover_1.hoverFocus)(enableHoverMediaQuery, "background-color: ".concat((0, _color.default)(bgColor).darken(0.2).rgb(), " !important;")), "\n    }\n  ");
}
var _default = {
  bgVariant: bgVariant
};
exports.default = _default;
});
unwrapExports(backgroundVariant);
var backgroundVariant_1 = backgroundVariant.bgVariant;
var backgroundVariant_2 = backgroundVariant.defaultProps;

var defaultProps = {
  '$enable-hover-media-query': false,
  '$brand-primary': '#0275d8',
  '$brand-success': '#5cb85c',
  '$brand-info': '#5bc0de',
  '$brand-warning': '#f0ad4e',
  '$brand-danger': '#d9543f',
  '$brand-inverse': '#373a3c',
  '$gray-lightest': '#f7f7f9'
};
function getBackgroundUtilities() {
  var $enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var $brandPrimary = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-primary'];
  var $brandSuccess = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps['$brand-success'];
  var $brandInfo = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : defaultProps['$brand-info'];
  var $brandWarning = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : defaultProps['$brand-warning'];
  var $brandDanger = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : defaultProps['$brand-danger'];
  var $brandInverse = arguments.length > 6 && arguments[6] !== undefined ? arguments[6] : defaultProps['$brand-inverse'];
  var $grayLightest = arguments.length > 7 && arguments[7] !== undefined ? arguments[7] : defaultProps['$gray-lightest'];
  return "\n    ".concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-primary', $brandPrimary), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-success', $brandSuccess), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-info', $brandInfo), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-warning', $brandWarning), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-danger', $brandDanger), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-inverse', $brandInverse), "\n    ").concat(backgroundVariant_1($enableHoverMediaQuery, '.bg-faded', $grayLightest), "\n  ");
}
var bgPrimary = function bgPrimary($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-primary'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-primary', bgColor);
};
var bgSuccess = function bgSuccess($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-success'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-success', bgColor);
};
var bgInfo = function bgInfo($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-info'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-info', bgColor);
};
var bgWarning = function bgWarning($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-warning'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-warning', bgColor);
};
var bgDanger = function bgDanger($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-danger'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-danger', bgColor);
};
var bgInverse = function bgInverse($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$brand-inverse'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-inverse', bgColor);
};
var bgFaded = function bgFaded($enableHoverMediaQuery) {
  var bgColor = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$gray-lightest'];
  return backgroundVariant_1($enableHoverMediaQuery, '.bg-faded', bgColor);
};
var background = {
  defaultProps: defaultProps,
  getBackgroundUtilities: getBackgroundUtilities,
  bgFaded: bgFaded,
  bgPrimary: bgPrimary,
  bgSuccess: bgSuccess,
  bgInfo: bgInfo,
  bgWarning: bgWarning,
  bgDanger: bgDanger,
  bgInverse: bgInverse
};

var borderRadius_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.borderRadius = borderRadius;
exports.borderTopRadius = borderTopRadius;
exports.borderRightRadius = borderRightRadius;
exports.borderBottomRadius = borderBottomRadius;
exports.borderLeftRadius = borderLeftRadius;
exports.default = exports.defaultProps = void 0;
var defaultProps = {
  '$border-radius': '.25rem',
  '$enable-rounded': true
};
exports.defaultProps = defaultProps;
function borderRadius() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$border-radius'];
  if (enableRounded) {
    return "\n      border-radius: ".concat(radius, ";\n    ");
  }
  return '';
}
function borderTopRadius() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$border-radius'];
  if (enableRounded) {
    return "\n      border-top-right-radius: ".concat(radius, ";\n      border-top-left-radius: ").concat(radius, ";\n    ");
  }
  return '';
}
function borderRightRadius() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$border-radius'];
  if (enableRounded) {
    return "\n      border-bottom-right-radius: ".concat(radius, ";\n      border-top-right-radius: ").concat(radius, ";\n    ");
  }
  return '';
}
function borderBottomRadius() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$border-radius'];
  if (enableRounded) {
    return "\n      border-bottom-right-radius: ".concat(radius, ";\n      border-bottom-left-radius: ").concat(radius, ";\n    ");
  }
  return '';
}
function borderLeftRadius() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$border-radius'];
  if (enableRounded) {
    return "\n      border-bottom-left-radius: ".concat(radius, ";\n      border-top-left-radius: ").concat(radius, ";\n    ");
  }
  return '';
}
var _default = {
  defaultProps: defaultProps,
  all: borderRadius,
  top: borderTopRadius,
  right: borderRightRadius,
  bottom: borderBottomRadius,
  left: borderLeftRadius
};
exports.default = _default;
});
unwrapExports(borderRadius_1);
var borderRadius_2 = borderRadius_1.borderRadius;
var borderRadius_3 = borderRadius_1.borderTopRadius;
var borderRadius_4 = borderRadius_1.borderRightRadius;
var borderRadius_5 = borderRadius_1.borderBottomRadius;
var borderRadius_6 = borderRadius_1.borderLeftRadius;
var borderRadius_7 = borderRadius_1.defaultProps;

var defaultProps$1 = {
  '$border-radius': '.25rem',
  '$enable-rounded': true
};
function getBordersUtilities() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    ".concat(rounded(enableRounded, radius), "\n    ").concat(roundedTop(enableRounded, radius), "\n    ").concat(roundedRight(enableRounded, radius), "\n    ").concat(roundedBottom(enableRounded, radius), "\n    ").concat(roundedLeft(enableRounded, radius), "\n    ").concat(roundedCircle(), "\n    ").concat(resetRounded(), "\n    ").concat(resetRoundedTop(), "\n    ").concat(resetRoundedRight(), "\n    ").concat(resetRoundedLeft(), "\n    ").concat(resetRoundedBottom(), "\n    ").concat(resetBorder(), "\n    ").concat(resetBorderTop(), "\n    ").concat(resetBorderRight(), "\n    ").concat(resetBorderLeft(), "\n    ").concat(resetBorderBottom(), "\n  ");
}
function rounded() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    .rounded {\n      ".concat(borderRadius_2(enableRounded, radius), "\n    }\n  ");
}
function roundedTop() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    .rounded-top {\n      ".concat(borderRadius_3(enableRounded, radius), "\n    }\n  ");
}
function roundedRight() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    .rounded-right {\n      ".concat(borderRadius_4(enableRounded, radius), "\n    }\n  ");
}
function roundedBottom() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    .rounded-bottom {\n      ".concat(borderRadius_5(enableRounded, radius), "\n    }\n  ");
}
function roundedLeft() {
  var enableRounded = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$1['$enable-rounded'];
  var radius = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$1['$border-radius'];
  return "\n    .rounded-left {\n      ".concat(borderRadius_6(enableRounded, radius), "\n    }\n  ");
}
function roundedCircle() {
  return "\n    .rounded-circle {\n      border-radius: 50%;\n    }\n  ";
}
function resetBorder() {
  return "\n    .border-0 {\n      border: 0 !important;\n    }\n  ";
}
function resetBorderTop() {
  return "\n    .border-top-0 {\n      border-top: 0 !important;\n    }\n  ";
}
function resetBorderRight() {
  return "\n    .border-right-0 {\n      border-right: 0 !important;\n    }\n  ";
}
function resetBorderBottom() {
  return "\n    .border-bottom-0 {\n      border-bottom: 0 !important;\n    }\n  ";
}
function resetBorderLeft() {
  return "\n    .border-left-0 {\n      border-left: 0 !important;\n    }\n  ";
}
function resetRounded() {
  return "\n    .rounded-0 {\n      border-radius: 0 !important;\n    }\n  ";
}
function resetRoundedTop() {
  return "\n    .rounded-top-0 {\n      border-top-left-radius: 0 !important;\n      border-top-right-radius: 0 !important;\n    }\n  ";
}
function resetRoundedBottom() {
  return "\n    .rounded-bottom-0 {\n      border-bottom-left-radius: 0 !important;\n      border-bottom-right-radius: 0 !important;\n    }\n  ";
}
function resetRoundedLeft() {
  return "\n    .rounded-left-0 {\n      border-bottom-left-radius: 0 !important;\n      border-top-left-radius: 0 !important;\n    }\n  ";
}
function resetRoundedRight() {
  return "\n    .rounded-right-0 {\n      border-bottom-right-radius: 0 !important;\n      border-top-right-radius: 0 !important;\n    }\n  ";
}
var borders = {
  defaultProps: defaultProps$1,
  getBordersUtilities: getBordersUtilities,
  rounded: rounded,
  roundedTop: roundedTop,
  roundedRight: roundedRight,
  roundedBottom: roundedBottom,
  roundedLeft: roundedLeft,
  roundedCircle: roundedCircle
};

var clearfix_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.clearfix = clearfix;
exports.default = void 0;
function clearfix() {
  return "\n    &::after {\n      content: \"\";\n      display: table;\n      clear: both;\n    }\n  ";
}
var _default = {
  clearfix: clearfix
};
exports.default = _default;
});
unwrapExports(clearfix_1);
var clearfix_2 = clearfix_1.clearfix;

function getClearfixUtilities() {
  return "\n   ".concat(getClearfix(), "\n  ");
}
function getClearfix() {
  return "\n    .clearfix {\n      ".concat(clearfix_2(), "\n    }\n  ");
}
var clearfix = {
  getClearfixUtilities: getClearfixUtilities,
  getClearfix: getClearfix
};

function getCursorUtilities() {
  return "\n    .cursor-alias {\n      cursor: alias;\n    }\n    \n    .cursor-all-scroll {\n      cursor: all-scroll;\n    }\n    \n    .cursor-auto {\n      cursor: auto;\n    }\n    \n    .cursor-cell {\n      cursor: cell;\n    }\n    \n    .cursor-context-menu {\n      cursor: context-menu;\n    }\n    \n    .cursor-col-resize {\n      cursor: col-resize;\n    }\n    \n    .cursor-copy {\n      cursor: copy;\n    }\n    \n    .cursor-crosshair {\n      cursor: crosshair;\n    }\n    \n    .cursor-default {\n      cursor: default;\n    }\n    \n    .cursor-e-resize {\n      cursor: e-resize;\n    }\n    \n    .cursor-ew-resize {\n      cursor: ew-resize;\n    }\n    \n    .cursor-grab {\n      cursor: grab;\n    }\n    \n    .cursor-grabbing {\n      cursor: grabbing;\n    }\n    \n    .cursor-help {\n      cursor: help;\n    }\n    \n    .cursor-move {\n      cursor: move;\n    }\n    \n    .cursor-n-resize {\n      cursor: n-resize;\n    }\n    \n    .cursor-ne-resize {\n      cursor: ne-resize;\n    }\n    \n    .cursor-nesw-resize {\n      cursor: nesw-resize;\n    }\n    \n    .cursor-ns-resize {\n      cursor: ns-resize;\n    }\n    \n    .cursor-nw-resize {\n      cursor: nw-resize;\n    }\n    \n    .cursor-nwse-resize {\n      cursor: nwse-resize;\n    }\n    \n    .cursor-no-drop {\n      cursor: no-drop;\n    }\n    \n    .cursor-none {\n      cursor: none;\n    }\n    \n    .cursor-not-allowed {\n      cursor: not-allowed;\n    }\n    \n    .cursor-pointer {\n      cursor: pointer;\n    }\n    \n    .cursor-progress {\n      cursor: progress;\n    }\n    \n    .cursor-row-resize {\n      cursor: row-resize;\n    }\n    \n    .cursor-s-resize {\n      cursor: s-resize;\n    }\n    \n    .cursor-se-resize {\n      cursor: se-resize;\n    }\n    \n    .cursor-sw-resize {\n      cursor: sw-resize;\n    }\n    \n    .cursor-text {\n      cursor: text;\n    }\n    \n    .cursor-vertical-text {\n      cursor: vertical-text;\n    }\n    \n    .cursor-w-resize {\n      cursor: w-resize;\n    }\n    \n    .cursor-wait {\n      cursor: wait;\n    }\n    \n    .cursor-zoom-in {\n      cursor: zoom-in;\n    }\n    \n    .cursor-zoom-out {\n      cursor: zoom-out;\n    }\n    \n    .cursor-initial {\n      cursor: initial;\n    }\n  ";
}
var cursor = {
  getCursorUtilities: getCursorUtilities
};

var unitUtils = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
var UnitUtils = function UnitUtils() {
  var _this = this;
  _classCallCheck(this, UnitUtils);
  this.UNIT = {
    EM: 'em',
    REM: 'rem',
    PX: 'px',
    PERCENT: '%'
  };
  this.math = {
    addition: function addition(a, b) {
      var unit = this.detectUnit(a) || this.detectUnit(b);
      return this.rmUnit(a) + this.rmUnit(b) + unit;
    }.bind(this),
    subtract: function subtract(a, b) {
      var unit = this.detectUnit(a) || this.detectUnit(b);
      return this.rmUnit(a) - this.rmUnit(b) + unit;
    }.bind(this),
    multiply: function multiply(a, b) {
      var unit = this.detectUnit(a) || this.detectUnit(b);
      return this.rmUnit(a) * this.rmUnit(b) + unit;
    }.bind(this),
    divide: function divide(a, b) {
      var unit = this.detectUnit(a) || this.detectUnit(b);
      return this.rmUnit(a) / this.rmUnit(b) + unit;
    }.bind(this)
  };
  this.detectUnit = function (value) {
    var ext;
    var valueStr = value.toString();
    if (valueStr.match(_this.UNIT.PX)) {
      ext = _this.UNIT.PX;
    } else if (valueStr.match(_this.UNIT.REM)) {
      ext = _this.UNIT.REM;
    } else if (valueStr.match(_this.UNIT.EM)) {
      ext = _this.UNIT.EM;
    } else if (valueStr.match(_this.UNIT.PERCENT)) {
      ext = _this.UNIT.PERCENT;
    } else if (!isNaN(value)) {
      return null;
    } else {
      throw new Error("detectUnit can't find unit for ".concat(value));
    }
    return ext;
  };
  this.rmUnit = function (value, unit) {
    var valueStr = value.toString();
    var ext = unit || _this.detectUnit(valueStr);
    var number = valueStr.replace(ext, '');
    return parseFloat(number);
  };
  this.toPercent = function (value) {
    var total = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 100;
    var decimal = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 2;
    return "".concat(Math.floor(value / total * 100 * Math.pow(10, decimal)) / Math.pow(10, decimal)).concat(_this.UNIT.PERCENT);
  };
};
var _default = new UnitUtils();
exports.default = _default;
module.exports = exports.default;
});
var index = unwrapExports(unitUtils);

var breakpoints = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.breakpointNext = breakpointNext;
exports.breakpointMin = breakpointMin;
exports.breakpointMax = breakpointMax;
exports.breakpointInfix = breakpointInfix;
exports.mediaBreakpointUp = mediaBreakpointUp;
exports.mediaBreakpointDown = mediaBreakpointDown;
exports.mediaBreakpointBetween = mediaBreakpointBetween;
exports.mediaBreakpointOnly = mediaBreakpointOnly;
exports.default = exports.defaultProps = void 0;
var _unitUtils = _interopRequireDefault(unitUtils);
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var defaultProps = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  }
};
exports.defaultProps = defaultProps;
function breakpointNext(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var breakpointNames = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : Object.keys(breakpoints);
  var n = breakpointNames.indexOf(name);
  if (n !== -1 && n + 1 < breakpointNames.length) {
    return breakpointNames[n + 1];
  }
  return null;
}
function breakpointMin(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var min = breakpoints[name];
  return min !== '0' ? min : null;
}
function breakpointMax(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var next = breakpointNext(name, breakpoints);
  if (next) {
    var min = _unitUtils.default.rmUnit(breakpointMin(next, breakpoints), _unitUtils.default.UNIT.PX);
    return (min - 1).toString() + _unitUtils.default.UNIT.PX;
  }
  return null;
}
function breakpointInfix(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  return !name || breakpointMin(name, breakpoints) === null ? '' : "-".concat(name);
}
function mediaBreakpointUp(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var content = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : '';
  var min = breakpointMin(name, breakpoints);
  if (min) {
    return "\n      @media (min-width: ".concat(min, ") {\n        ").concat(content, "\n      }\n    ");
  }
  return content;
}
function mediaBreakpointDown(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var content = arguments.length > 2 ? arguments[2] : undefined;
  var max = breakpointMax(name, breakpoints);
  if (max) {
    return "\n      @media (max-width: ".concat(max, ") {\n        ").concat(content, "\n      }\n    ");
  }
  return content;
}
function mediaBreakpointBetween(lower, upper) {
  var breakpoints = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps['$grid-breakpoints'];
  var content = arguments.length > 3 ? arguments[3] : undefined;
  var min = breakpointMin(lower, breakpoints);
  var max = breakpointMax(upper, breakpoints);
  if (min && max) {
    return "\n      @media (min-width: ".concat(min, ") and (max-width: ").concat(max, ") {\n        ").concat(content, "\n      }\n    ");
  } else if (min) {
    return "\n      @media (min-width: ".concat(min, ") {\n        ").concat(content, "\n      }\n    ");
  } else if (max) {
    return "\n      @media (max-width: ".concat(max, ") {\n        ").concat(content, "\n      }\n    ");
  }
  return content;
}
function mediaBreakpointOnly(name) {
  var breakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps['$grid-breakpoints'];
  var content = arguments.length > 2 ? arguments[2] : undefined;
  return mediaBreakpointBetween(name, name, breakpoints, content);
}
var _default = {
  defaultProps: defaultProps,
  up: mediaBreakpointUp,
  down: mediaBreakpointDown,
  between: mediaBreakpointBetween,
  only: mediaBreakpointOnly
};
exports.default = _default;
});
unwrapExports(breakpoints);
var breakpoints_1 = breakpoints.breakpointNext;
var breakpoints_2 = breakpoints.breakpointMin;
var breakpoints_3 = breakpoints.breakpointMax;
var breakpoints_4 = breakpoints.breakpointInfix;
var breakpoints_5 = breakpoints.mediaBreakpointUp;
var breakpoints_6 = breakpoints.mediaBreakpointDown;
var breakpoints_7 = breakpoints.mediaBreakpointBetween;
var breakpoints_8 = breakpoints.mediaBreakpointOnly;
var breakpoints_9 = breakpoints.defaultProps;

var defaultProps$2 = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  }
};
function getDisplayUtilities() {
  var gridBreakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$2['$grid-breakpoints'];
  var utilityList = [];
  Object.keys(gridBreakpoints).forEach(function (breakpoint) {
    var infix = breakpoints_4(breakpoint, gridBreakpoints);
    utilityList.push("\n      ".concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .d".concat(infix, "-none { display: none !important; }\n        .d").concat(infix, "-inline { display: inline !important; }\n        .d").concat(infix, "-inline-block { display: inline-block !important; }\n        .d").concat(infix, "-block { display: block !important; }\n        .d").concat(infix, "-table { display: table !important; }\n        .d").concat(infix, "-table-cell { display: table-cell !important; }\n        .d").concat(infix, "-flex { display: flex !important; }\n        .d").concat(infix, "-inline-flex { display: inline-flex !important; }\n      ")), "\n    "));
  });
  utilityList.push("\n    .d-print-block {\n      display: none !important;\n    \n      @media print {\n        display: block !important;\n      }\n    }\n    \n    .d-print-inline {\n      display: none !important;\n    \n      @media print {\n        display: inline !important;\n      }\n    }\n    \n    .d-print-inline-block {\n      display: none !important;\n    \n      @media print {\n        display: inline-block !important;\n      }\n    }\n    \n    .d-print-none {\n      @media print {\n        display: none !important;\n      }\n    }\n  ");
  return utilityList.join('\n');
}
var display = {
  defaultProps: defaultProps$2,
  getDisplayUtilities: getDisplayUtilities
};

var defaultProps$3 = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  }
};
function getFlexUtilities() {
  var gridBreakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$3['$grid-breakpoints'];
  var flexUtilityList = [];
  Object.keys(gridBreakpoints).forEach(function (breakpoint) {
    var infix = breakpoints_4(breakpoint, gridBreakpoints);
    flexUtilityList.push("\n      /* Flex column reordering */\n      ".concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .flex".concat(infix, "-first { order: -1; }\n        .flex").concat(infix, "-last { order: 1; }\n        .flex").concat(infix, "-unordered { order: 0; }\n      ")), "\n  \n      /* Flex direction */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .flex".concat(infix, "-row            { flex-direction: row !important; }\n        .flex").concat(infix, "-column         { flex-direction: column !important; }\n        .flex").concat(infix, "-row-reverse    { flex-direction: row-reverse !important; }\n        .flex").concat(infix, "-column-reverse { flex-direction: column-reverse !important; }\n      ")), "\n      \n      /* Flex wrap */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .flex".concat(infix, "-wrap         { flex-wrap: wrap !important; }\n        .flex").concat(infix, "-nowrap       { flex-wrap: nowrap !important; }\n        .flex").concat(infix, "-wrap-reverse { flex-wrap: wrap-reverse !important; }\n      ")), "\n      /* Flex justify-content */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .justify-content".concat(infix, "-start   { justify-content: flex-start !important; }\n        .justify-content").concat(infix, "-end     { justify-content: flex-end !important; }\n        .justify-content").concat(infix, "-center  { justify-content: center !important; }\n        .justify-content").concat(infix, "-between { justify-content: space-between !important; }\n        .justify-content").concat(infix, "-around  { justify-content: space-around !important; }\n        .justify-content").concat(infix, "-evenly  { justify-content: space-evenly !important; }\n      ")), "\n      /* Flex align-items */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .align-items".concat(infix, "-start    { align-items: flex-start !important; }\n        .align-items").concat(infix, "-end      { align-items: flex-end !important; }\n        .align-items").concat(infix, "-center   { align-items: center !important; }\n        .align-items").concat(infix, "-baseline { align-items: baseline !important; }\n        .align-items").concat(infix, "-stretch  { align-items: stretch !important; }\n      ")), "\n      /* Flex align-content */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .align-content".concat(infix, "-start   { align-content: flex-start !important; }\n        .align-content").concat(infix, "-end     { align-content: flex-end !important; }\n        .align-content").concat(infix, "-center  { align-content: center !important; }\n        .align-content").concat(infix, "-between { align-content: space-between !important; }\n        .align-content").concat(infix, "-around  { align-content: space-around !important; }\n        .align-content").concat(infix, "-stretch { align-content: stretch !important; }\n      ")), "\n      /* Flex align-self */ \n      ").concat(breakpoints_5(breakpoint, gridBreakpoints, "\n        .align-self".concat(infix, "-auto     { align-self: auto !important; }\n        .align-self").concat(infix, "-start    { align-self: flex-start !important; }\n        .align-self").concat(infix, "-end      { align-self: flex-end !important; }\n        .align-self").concat(infix, "-center   { align-self: center !important; }\n        .align-self").concat(infix, "-baseline { align-self: baseline !important; }\n        .align-self").concat(infix, "-stretch  { align-self: stretch !important; }\n      ")), " \n    "));
  });
  return flexUtilityList.join('\n');
}
var flex = {
  defaultProps: defaultProps$3,
  getFlexUtilities: getFlexUtilities
};

var float_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.floatLeft = floatLeft;
exports.floatRight = floatRight;
exports.floatNone = floatNone;
exports.default = void 0;
function floatLeft() {
  return "\n    float: left !important;\n  ";
}
function floatRight() {
  return "\n    float: right !important;\n  ";
}
function floatNone() {
  return "\n    float: none !important;\n  ";
}
var _default = {
  floatLeft: floatLeft,
  floatRight: floatRight,
  floatNone: floatNone
};
exports.default = _default;
});
unwrapExports(float_1);
var float_2 = float_1.floatLeft;
var float_3 = float_1.floatRight;
var float_4 = float_1.floatNone;

var defaultProps$4 = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  }
};
function getFloatUtilities() {
  var gridBreakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$4['$grid-breakpoints'];
  var floatUtilityList = [];
  Object.keys(gridBreakpoints).forEach(function (breakpoint) {
    var infix = breakpoints_4(breakpoint, gridBreakpoints);
    var floatUtility = breakpoints_5(breakpoint, gridBreakpoints, "\n      .float".concat(infix, "-left {\n        ").concat(float_2(), "\n      }\n      .float").concat(infix, "-right {\n        ").concat(float_3(), "\n      }\n      .float").concat(infix, "-none {\n        ").concat(float_4(), "\n      }\n    "));
    floatUtilityList.push(floatUtility);
  });
  return floatUtilityList.join('\n');
}
var float = {
  defaultProps: defaultProps$4,
  getFloatUtilities: getFloatUtilities
};

var defaultProps$5 = {
  '$zindex-fixed': '1030',
  '$zindex-sticky': '1030'
};
function getPositionUtilities() {
  var zindexFixed = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$5['$zindex-fixed'];
  var zindexSticky = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$5['$zindex-sticky'];
  return "\n    ".concat(fixedTop(zindexFixed), "\n    ").concat(fixedBottom(zindexFixed), "\n    ").concat(stickTop(zindexSticky), "\n  ");
}
function fixedTop() {
  var zindexFixed = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$5['$zindex-fixed'];
  return "\n  .fixed-top {\n    position: fixed !important;\n    top: 0;\n    right: 0;\n    left: 0;\n    z-index: ".concat(zindexFixed, ";\n  }\n  ");
}
function fixedBottom() {
  var zindexFixed = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$5['$zindex-fixed'];
  return "\n    .fixed-bottom {\n      position: fixed !important;\n      right: 0;\n      bottom: 0;\n      left: 0;\n      z-index: ".concat(zindexFixed, ";\n    }\n  ");
}
function stickTop() {
  var zindexSticky = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$5['$zindex-sticky'];
  return "\n    .sticky-top {\n      position: sticky !important;\n      top: 0;\n      z-index: ".concat(zindexSticky, ";\n    }\n  ");
}
var position = {
  defaultProps: defaultProps$5,
  getPositionUtilities: getPositionUtilities,
  fixedTop: fixedTop,
  fixedBottom: fixedBottom,
  stickTop: stickTop
};

var defaultProps$6 = {
  '$font-family-base': '-apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
  '$font-size-base': '1rem',
  '$font-weight-base': '1.5',
  '$line-height-base': '1.5',
  '$body-color': '#292b2c',
  '$body-bg': '#fff'
};
function getGlobalStyles() {
  return "\n    html {\n      ".concat(html(), "\n    }\n    *,\n    *::before,\n    *::after {\n      ").concat(boxSizing(), "\n    }\n    @-ms-viewport { \n      ").concat(ie10FixViewport(), " \n    }\n    body {\n      ").concat(bodyUtils(), "\n    }\n  ");
}
function getGlobalStyleNoBootstrapProvider() {
  var fontFamilyBase = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$6['$font-family-base'];
  var fontSizeBase = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$6['$font-size-base'];
  var fontWeightBase = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps$6['$font-weight-base'];
  var lineHeightBase = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : defaultProps$6['$line-height-base'];
  var bodyColor = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : defaultProps$6['$body-color'];
  var bodyBg = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : defaultProps$6['$body-bg'];
  return "\n  ".concat(getGlobalStyles(), "\n  body {\n  ").concat(body(fontFamilyBase, fontSizeBase, fontWeightBase, lineHeightBase, bodyColor, bodyBg), " \n}");
}
function html() {
  return "\n    box-sizing: border-box;\n    font-family: sans-serif;\n    line-height: 1.15;\n    -ms-text-size-adjust: 100%;\n    -webkit-text-size-adjust: 100%;\n    -ms-overflow-style: scrollbar;\n    -webkit-tap-highlight-color: rgba(0,0,0,0);\n  ";
}
function boxSizing() {
  return "\n    box-sizing: inherit;\n  ";
}
function ie10FixViewport() {
  return "\n    width: device-width;\n  ";
}
function body() {
  var fontFamilyBase = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$6['$font-family-base'];
  var fontSizeBase = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$6['$font-size-base'];
  var fontWeightBase = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps$6['$font-weight-base'];
  var lineHeightBase = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : defaultProps$6['$line-height-base'];
  var bodyColor = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : defaultProps$6['$body-color'];
  var bodyBg = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : defaultProps$6['$body-bg'];
  return "\n    margin: 0;\n    font-family: ".concat(fontFamilyBase, ";\n    font-size: ").concat(fontSizeBase, ";\n    font-weight: ").concat(fontWeightBase, ";\n    line-height: ").concat(lineHeightBase, ";\n    color: ").concat(bodyColor, ";\n    background-color: ").concat(bodyBg, ";\n    \n    ").concat(bodyUtils(), "\n    \n    [tabindex=\"-1\"]:focus {\n      ").concat(tabIndex(), "\n    }\n    svg:not(:root) {\n      ").concat(svg(), "\n    }\n    [hidden] {\n      ").concat(ie10FixHidden(), "\n    }\n    ::-webkit-file-upload-button {\n      ").concat(webkitFileUploadButton(), "\n    }\n  ");
}
function bodyUtils() {
  return "\n    &.overflow {\n      overflow: hidden;\n    }\n  ";
}
function tabIndex() {
  return "\n    outline: none !important;\n  ";
}
function svg() {
  return "\n    overflow: hidden;\n  ";
}
function ie10FixHidden() {
  return "\n    display: none !important;\n  ";
}
function webkitFileUploadButton() {
  return "\n    font: inherit;\n    -webkit-appearance: button;\n  ";
}
var reboot = {
  html: html,
  boxSizing: boxSizing,
  ie10FixViewport: ie10FixViewport,
  body: body,
  bodyUtils: bodyUtils,
  tabIndex: tabIndex,
  svg: svg,
  ie10FixHidden: ie10FixHidden,
  getGlobalStyles: getGlobalStyles,
  getGlobalStyleNoBootstrapProvider: getGlobalStyleNoBootstrapProvider,
  webkitFileUploadButton: webkitFileUploadButton
};

var screenReader = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.srOnly = srOnly;
exports.srOnlyFocusable = srOnlyFocusable;
exports.default = void 0;
function srOnly() {
  return "\n    position: absolute !important;\n    width: 1px;\n    height: 1px;\n    padding: 0;\n    margin: -1px;\n    overflow: hidden;\n    clip: rect(0,0,0,0);\n    border: 0;\n  ";
}
function srOnlyFocusable() {
  return "\n    &:active,\n    &:focus {\n      position: static;\n      width: auto;\n      height: auto;\n      margin: 0;\n      overflow: visible;\n      clip: auto;\n    }\n  ";
}
var _default = {
  srOnly: srOnly,
  srOnlyFocusable: srOnlyFocusable
};
exports.default = _default;
});
unwrapExports(screenReader);
var screenReader_1 = screenReader.srOnly;
var screenReader_2 = screenReader.srOnlyFocusable;

function getScreenReadersUtilities() {
  return "\n    .sr-only {\n      ".concat(screenReader_1(), "\n    }\n    \n    .sr-only-focusable {\n      ").concat(screenReader_2(), "\n    }\n  ");
}
var screenreaders = {
  getScreenReadersUtilities: getScreenReadersUtilities
};

var defaultProps$7 = {
  $sizes: {
    25: '25%',
    50: '50%',
    75: '75%',
    100: '100%'
  }
};
function getSizingUtilities() {
  var sizes = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$7['$sizes'];
  var abbrev = {
    width: 'w',
    height: 'h'
  };
  var sizingList = [];
  Object.keys(abbrev).forEach(function (cssProp) {
    Object.keys(sizes).forEach(function (size) {
      sizingList.push("\n        .".concat(abbrev[cssProp], "-").concat(size, " { ").concat(cssProp, ": ").concat(sizes[size], " !important; }\n      "));
    });
  });
  return "\n    ".concat(sizingList.join('\n'), "\n    .mw-100 { max-width: 100% !important; }\n    .mh-100 { max-height: 100% !important; }\n  ");
}
var sizing = {
  defaultProps: defaultProps$7,
  getSizingUtilities: getSizingUtilities
};

var defaultProps$8 = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  },
  '$spacers': {
    0: {
      x: 0,
      y: 0
    },
    1: {
      x: '0.25rem',
      y: '0.25rem'
    },
    2: {
      x: '0.5rem',
      y: '0.5rem'
    },
    3: {
      x: '1rem',
      y: '1rem'
    },
    4: {
      x: '1.5rem',
      y: '1.5rem'
    },
    5: {
      x: '3rem',
      y: '3rem'
    }
  }
};
function getSpacingUtilities()
{
  var gridBreakpoints = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$8['$grid-breakpoints'];
  var spacers = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$8['$spacers'];
  var abbrevs = {
    margin: 'm',
    padding: 'p'
  };
  var spacingUtilityList = [];
  var infixList = [];
  Object.keys(gridBreakpoints).forEach(function (breakpoint) {
    var infix = breakpoints_4(breakpoint, gridBreakpoints);
    infixList.push(infix);
    Object.keys(abbrevs).forEach(function (prop) {
      var abbrev = abbrevs[prop];
      Object.keys(spacers).forEach(function (size) {
        var lengths = spacers[size];
        spacingUtilityList.push(breakpoints_5(breakpoint, gridBreakpoints, "\n          .".concat(abbrev).concat(infix, "-").concat(size, " { ").concat(prop, ":        ").concat(lengths.y, " ").concat(lengths.x, " !important; } /* a = All sides */\n          .").concat(abbrev, "t").concat(infix, "-").concat(size, " { ").concat(prop, "-top:    ").concat(lengths.y, " !important; }\n          .").concat(abbrev, "r").concat(infix, "-").concat(size, " { ").concat(prop, "-right:  ").concat(lengths.x, " !important; }\n          .").concat(abbrev, "b").concat(infix, "-").concat(size, " { ").concat(prop, "-bottom: ").concat(lengths.y, " !important; }\n          .").concat(abbrev, "l").concat(infix, "-").concat(size, " { ").concat(prop, "-left:   ").concat(lengths.x, " !important; }\n          .").concat(abbrev, "x").concat(infix, "-").concat(size, " {\n            ").concat(prop, "-right:  ").concat(lengths.x, " !important;\n            ").concat(prop, "-left:   ").concat(lengths.x, " !important;\n          }\n          .").concat(abbrev, "y").concat(infix, "-").concat(size, " {\n            ").concat(prop, "-top:    ").concat(lengths.y, " !important;\n            ").concat(prop, "-bottom: ").concat(lengths.y, " !important;\n          }\n        ")));
      });
    });
  });
  var infixUtilityList = infixList.map(function (infix) {
    return "\n    .m".concat(infix, "-auto  { margin:        auto !important; }\n    .mt").concat(infix, "-auto { margin-top:    auto !important; }\n    .mr").concat(infix, "-auto { margin-right:  auto !important; }\n    .mb").concat(infix, "-auto { margin-bottom: auto !important; }\n    .ml").concat(infix, "-auto { margin-left:   auto !important; }\n    .mx").concat(infix, "-auto {\n      margin-right: auto !important;\n      margin-left:  auto !important;\n    }\n    .my").concat(infix, "-auto {\n      margin-top:    auto !important;\n      margin-bottom: auto !important;\n    }\n  ");
  });
  return "\n    ".concat(infixUtilityList.join('\n'), "\n    ").concat(spacingUtilityList.join('\n'), "\n  ");
}
var spacing = {
  defaultProps: defaultProps$8,
  getSpacingUtilities: getSpacingUtilities
};

var parseTransition_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var parseTransition = function parseTransition(transitions) {
  if (!transitions) {
    return [];
  }
  var sample = transitions;
  var RULE_DELIMITER = ',';
  var PROPERTY_DELIMITER = ' ';
  var MS_UNIT = 'ms';
  var TMP_STR = 'TMP';
  var DEFAULT_PROPERTY = 'all';
  var DEFAULT_DURATION = 0;
  var DEFAULT_TIMING_FUNCTION = 'ease';
  var DEFAULT_DELAY = 0;
  var BEZIER_REGEX = /cubic-bezier\([^\)]+\)/gi;
  var cubicBezierList = transitions.match(BEZIER_REGEX);
  if (cubicBezierList) {
    sample = sample.replace(BEZIER_REGEX, TMP_STR);
  }
  var transitionList = sample.split(RULE_DELIMITER).map(function (rule) {
    var properties = rule.trim().split(PROPERTY_DELIMITER);
    return {
      property: properties[0] || DEFAULT_PROPERTY,
      duration: properties[1] && !(properties[1].indexOf(MS_UNIT) !== -1) ? parseFloat(properties[1]) * 1000 : parseFloat(properties[1]) || DEFAULT_DURATION,
      timingFunction: properties[2] && properties[2] !== TMP_STR ? properties[2] : cubicBezierList ? cubicBezierList.shift() : DEFAULT_TIMING_FUNCTION,
      delay: properties[3] && !(properties[3].indexOf(MS_UNIT) !== -1) ? parseFloat(properties[3]) * 1000 : parseFloat(properties[3]) || DEFAULT_DELAY
    };
  });
  return transitionList;
};
var _default = parseTransition;
exports.default = _default;
module.exports = exports.default;
});
var parseTransition = unwrapExports(parseTransition_1);

var transition_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transition = transition;
exports.default = exports.defaultProps = void 0;
var defaultProps = {
  '$enable-transitions': true
};
exports.defaultProps = defaultProps;
function transition() {
  var enableTransitions = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-transitions'];
  for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
    args[_key - 1] = arguments[_key];
  }
  if (enableTransitions && args.length) {
    return "\n      transition: ".concat(args.join(' '), ";\n    ");
  }
  return '';
}
var _default = {
  transition: transition
};
exports.default = _default;
});
unwrapExports(transition_1);
var transition_2 = transition_1.transition;
var transition_3 = transition_1.defaultProps;

var defaultProps$9 = {
  '$enable-transitions': true,
  '$transition-fade': 'opacity .15s linear',
  '$transition-collapse': 'height .35s ease'
};
function getTransitionUtilities() {
  var enableTransitions = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$9['$enable-transitions'];
  var transitionFade = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$9['$transition-fade'];
  var transitionCollapse = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps$9['$transition-collapse'];
  return "\n    ".concat(fade(enableTransitions, transitionFade), "\n    ").concat(collapse(enableTransitions, transitionCollapse), "\n  ");
}
function fade() {
  var enableTransitions = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$9['$enable-transitions'];
  var transitionFade = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$9['$transition-fade'];
  return "\n    .fade,\n     &.fade {\n      opacity: 0;\n      ".concat(transition_2(enableTransitions, transitionFade), "\n    \n      &.show {\n        opacity: 1;\n      }\n    }\n  ");
}
function collapse() {
  var enableTransitions = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$9['$enable-transitions'];
  var transitionCollapse = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$9['$transition-collapse'];
  return "\n    &.collapse, .collapse {\n      display: none;\n      &.show {\n        display: block;\n      }\n    }\n    \n    tr&, tr {\n      &.collapse.show {\n        display: table-row;\n      }\n    }\n    \n    tbody&, tbody {\n      &.collapse.show {\n        display: table-row-group;\n      }\n    }\n    \n    &.collapsing, .collapsing {\n      position: relative;\n      height: 0;\n      overflow: hidden;\n      ".concat(transition_2(enableTransitions, transitionCollapse), "\n    }\n  ");
}
function getReactTransition(enableTransition, transition) {
  var transitionList = parseTransition(transition);
  var _transitionList$ = transitionList[0],
      property = _transitionList$.property,
      duration = _transitionList$.duration,
      timingFunction = _transitionList$.timingFunction,
      delay = _transitionList$.delay;
  return transition_2(enableTransition, "".concat(property, " ").concat(duration, "ms ").concat(timingFunction, " ").concat(delay, "ms"));
}
var transition = {
  defaultProps: defaultProps$9,
  getTransitionUtilities: getTransitionUtilities,
  getReactTransition: getReactTransition,
  fade: fade,
  collapse: collapse
};

var textTruncate_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textTruncate = textTruncate;
exports.default = void 0;
function textTruncate() {
  return "\n    overflow: hidden;\n    text-overflow: ellipsis;\n    white-space: nowrap;\n  ";
}
var _default = {
  textTruncate: textTruncate
};
exports.default = _default;
});
unwrapExports(textTruncate_1);
var textTruncate_2 = textTruncate_1.textTruncate;

var textHide_1 = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textHide = textHide;
exports.default = void 0;
function textHide() {
  return "\n    font: 0/0 a;\n    color: transparent;\n    text-shadow: none;\n    background-color: transparent;\n    border: 0;\n  ";
}
var _default = {
  textHide: textHide
};
exports.default = _default;
});
unwrapExports(textHide_1);
var textHide_2 = textHide_1.textHide;

var textEmphasis = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.textEmphasisVariant = textEmphasisVariant;
exports.default = exports.defaultProps = void 0;
var _color = _interopRequireDefault(ColorWrapper);
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
var defaultProps = {
  '$enable-hover-media-query': false
};
exports.defaultProps = defaultProps;
function textEmphasisVariant() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps['$enable-hover-media-query'];
  var parent = arguments.length > 1 ? arguments[1] : undefined;
  var textColor = arguments.length > 2 ? arguments[2] : undefined;
  return "\n  ".concat(parent, " {\n    color: ").concat(textColor, " !important;\n  }\n  a").concat(parent, " {\n  ").concat((0, hover_1.hoverFocus)(enableHoverMediaQuery, "color: ".concat((0, _color.default)(textColor).darken(0.20).rgb(), " !important;")), "\n  }\n");
}
var _default = {
  defaultProps: defaultProps,
  textEmphasisVariant: textEmphasisVariant
};
exports.default = _default;
});
unwrapExports(textEmphasis);
var textEmphasis_1 = textEmphasis.textEmphasisVariant;
var textEmphasis_2 = textEmphasis.defaultProps;

var defaultProps$a = {
  '$grid-breakpoints': {
    xs: '0',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px'
  },
  '$enable-hover-media-query': false,
  '$font-weight-normal': 'normal',
  '$font-weight-bold': 'bold',
  '$text-muted': '#818a91',
  '$brand-primary': '#0275d8',
  '$brand-success': '#5cb85c',
  '$brand-info': '#5bc0de',
  '$brand-warning': '#f0ad4e',
  '$brand-danger': '#d9534f',
  '$gray-dark': '#373a3c'
};
function getTextUtilities() {
  var enableHoverMediaQuery = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : defaultProps$a['$enable-hover-media-query'];
  var gridBreakpoints = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : defaultProps$a['$grid-breakpoints'];
  var fontWeightNormal = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : defaultProps$a['$font-weight-normal'];
  var fontWeightBold = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : defaultProps$a['$font-weight-bold'];
  var textMuted = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : defaultProps$a['$text-muted'];
  var brandPrimary = arguments.length > 5 && arguments[5] !== undefined ? arguments[5] : defaultProps$a['$brand-primary'];
  var brandSuccess = arguments.length > 6 && arguments[6] !== undefined ? arguments[6] : defaultProps$a['$brand-success'];
  var brandInfo = arguments.length > 7 && arguments[7] !== undefined ? arguments[7] : defaultProps$a['$brand-info'];
  var brandWarning = arguments.length > 8 && arguments[8] !== undefined ? arguments[8] : defaultProps$a['$brand-warning'];
  var brandDanger = arguments.length > 9 && arguments[9] !== undefined ? arguments[9] : defaultProps$a['$brand-danger'];
  var grayDark = arguments.length > 10 && arguments[10] !== undefined ? arguments[10] : defaultProps$a['$gray-dark'];
  var responseAlignmentList = [];
  Object.keys(gridBreakpoints).forEach(function (bp) {
    var infix = breakpoints_4(bp, gridBreakpoints);
    var responsiveAlignement = breakpoints_5(bp, gridBreakpoints, "\n      .text".concat(infix, "-left { text-align: left !important; }\n      .text").concat(infix, "-right  { text-align: right !important; }\n      .text").concat(infix, "-center { text-align: center !important; }\n    "));
    responseAlignmentList.push(responsiveAlignement);
  });
  return "\n    /* Text */\n\n    /* Alignment */\n\n    .text-justify        { text-align: justify !important; }\n    .text-nowrap         { white-space: nowrap !important; }\n    .text-truncate       { ".concat(textTruncate_2(), " }\n\n    /* Responsive alignment */\n\n    ").concat(responseAlignmentList.join('\n'), "\n\n    /* Transformation */\n\n    .text-lowercase      { text-transform: lowercase !important; }\n    .text-uppercase      { text-transform: uppercase !important; }\n    .text-capitalize     { text-transform: capitalize !important; }\n\n    /* Weight and italics */\n\n    .font-weight-normal  { font-weight: ").concat(fontWeightNormal, "; }\n    .font-weight-bold    { font-weight: ").concat(fontWeightBold, "; }\n    .font-italic         { font-style: italic; }\n\n    /* Contextual colors */\n\n    .text-white {\n      color: #fff !important;\n    }\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-muted', textMuted), "\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-primary', brandPrimary), "\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-success', brandSuccess), "\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-info', brandInfo), "\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-warning', brandWarning), "\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-danger', brandDanger), "\n\n    /* Font color */\n\n    ").concat(textEmphasis_1(enableHoverMediaQuery, '.text-gray-dark', grayDark), "\n\n    /* Misc */\n\n    .text-hide {\n      ").concat(textHide_2(), "\n    }\n\n  ");
}
var text = {
  defaultProps: defaultProps$a,
  getTextUtilities: getTextUtilities
};

var visibility = createCommonjsModule(function (module, exports) {
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.invisible = invisible;
exports.default = void 0;
function invisible(visibility) {
  return "\n    visibility: ".concat(visibility, " !important;\n  ");
}
var _default = {
  invisible: invisible
};
exports.default = _default;
});
unwrapExports(visibility);
var visibility_1 = visibility.invisible;

function getVisibilityUtilities() {
  return "\n    .visible {\n      ".concat(visibility_1('visible'), "\n    }\n    \n    .invisible {\n      ").concat(visibility_1('hidden'), "\n    }\n   \n  ");
}
var visibility$1 = {
  getVisibilityUtilities: getVisibilityUtilities
};

export { align as alignUtils, background as backgroundUtils, borders as bordersUtils, clearfix as clearfixUtils, cursor as cursorUtils, display as displayUtils, flex as flexUtils, float as floatUtils, getGlobalStyleNoBootstrapProvider, getGlobalStyles, position as positionUtils, reboot as rebootUtils, screenreaders as screenreadersUtils, sizing as sizingUtils, spacing as spacingUtils, text as textUtils, transition as transitionUtils, index as unit, visibility$1 as visibilityUtils };
